osf/models/node.py

```diff
@@ -1230,9 +1230,16 @@ def set_privacy(self, permissions, auth=None, log=True, save=True, meeting_creat
         elif permissions == 'private' and self.is_public:
             if self.is_registration and not self.is_pending_embargo:
                 raise NodeStateError('Public registrations must be withdrawn, not made private.')
-            else:
-                self.is_public = False
-                self.keenio_read_key = ''
+
+            if auth:
+                if self.storage_limit_status is not None:
```

Comment:
nit: by making this `if storage_limit_status is None` you can put the else logic here and turn the nested if into an elif, reducing your complexity